controllers/clusterversion_controller.go

```diff
@@ -139,6 +141,32 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	// deletion phase
+	if !instance.GetDeletionTimestamp().IsZero() {
+		result, err := c.deletionPhase()
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if !result.Requeue {
+			//remove finalizer
+			controllerutil.RemoveFinalizer(&instance, ClusterVersionFinalizer)
+			if err := c.Client.Update(c.ctx, &instance); err != nil {
+				return ctrl.Result{}, err
+			}
+			c.log.Info("finallizer removed successfully")
+		}
+		return result, nil
+	}
+
+	//ensure finalizer
+	if !controllerutil.ContainsFinalizer(&instance, ClusterVersionFinalizer) {
```

Comment:
the way the code is written right now there is no clear code separation between deletion and the other reconciliation phases. can you please move everything after deletion phase handling into an else section?

or even better, reorder to the deletion phase handling code will be on the else path

```
 if instance.GetDeletionTimestamp().IsZero() {
     (normal reconcile)
 } else {
     (deletion reconcile)
 }

Reply:
i've tried to apply this approach in a previous iteration and got flagged by lint iirc, i'll try once more and update in case not possible.